rct229/rules/section22/section22rule25.py

```diff
@@ -0,0 +1,112 @@
+
+
+class Section22Rule25(RuleDefinitionListIndexedBase):
+    """Rule 25 of ASHRAE 90.1-2019 Appendix G Section 22 (Chilled water loop)"""
+
+    def __init__(self):
+        super(Section22Rule25, self).__init__(
+            rmrs_used=UserBaselineProposedVals(False, True, False),
+            each_rule=Section22Rule25.PrimaryCoolingFluidLoop(),
+            index_rmr="baseline",
+            id="22-25",
+            description="For chilled-water systems served by chiller(s) and does not serve baseline System-11, the baseline building constant-volume primary pump power shall be modeled as 9 W/gpm.",
+            rmr_context="ruleset_model_instances/0",
+            list_path="fluid_loops[*]",
+        )
+
+    def is_applicable(self, context, data=None):
+        rmi_b = context.baseline
+        baseline_system_types_dict = get_baseline_system_types(rmi_b)
+        # create a list contains all HVAC systems that are modeled in the rmi_b
+        available_type_lists = [
+            hvac_type
+            for hvac_type in baseline_system_types_dict.keys()
+            if len(baseline_system_types_dict[hvac_type]) > 0
+        ]
+        primary_secondary_loop_dict = get_primary_secondary_loops_dict(rmi_b)
+
+        return (
+            any(
+                [
+                    available_type in APPLICABLE_SYS_TYPES
+                    for available_type in available_type_lists
+                ]
+            )
+            and any(
+                [
+                    available_type not in NOT_APPLICABLE_SYS_TYPES
+                    for available_type in available_type_lists
+                ]
+            )
+            and primary_secondary_loop_dict
+        )
+
+    def create_data(self, context, data):
+        rmi_b = context.baseline
+        primary_secondary_loops_dict = get_primary_secondary_loops_dict(rmi_b)
+        return {"primary_secondary_loops_dict": primary_secondary_loops_dict}
+
+    def list_filter(self, context_item, data):
+        fluid_loop = context_item.baseline
+        primary_secondary_loops_dict = data["primary_secondary_loops_dict"]
+        return fluid_loop["id"] in primary_secondary_loops_dict.keys()
+
+    class PrimaryCoolingFluidLoop(RuleDefinitionBase):
+        def __init__(self):
+            super(Section22Rule25.PrimaryCoolingFluidLoop, self).__init__(
+                rmrs_used=UserBaselineProposedVals(False, True, False),
+                required_fields={
+                    "$": ["pump_power_per_flow_rate"],
+                },
+            )
+
+        def get_calc_vals(self, context, data=None):
+            primary_cooling_loop = context.baseline
+            primary_pump_power_per_flow_rate = primary_cooling_loop[
+                "pump_power_per_flow_rate"
+            ]
+            return {
+                "primary_pump_power_per_flow_rate": primary_pump_power_per_flow_rate,
+                "required_pump_power_per_flow_rate": REQUIRED_PUMP_POWER_PER_FLOW_RATE,
```

Comment:
Could you check whether we have consistently compared ```REQUIRED_PUMP_POWER_PER_FLOW_RATE``` in the ```rule_check``` method? This is correct but, I'm wondering what would be the consistent way to develop the code. Thanks!

Reply:
The reason I put it here was because I realized we need the `REQUIRED_PUMP_POWER_PER_FLOW_RATE` in the report. 
To note, only data in the `calc_val` will go into the final reporting. If we dont include this data in the calc_val output, then it missed in the ruleset checking output.
I suggest to follow this way instead of use it in the `rule_check` method.